File: java/class/duotai/Duotai.java
public class Duotai{
    public static void main(String[] args){
        Animal animal = new Animal(); 
        animal.say();
        animal = new Dog();
        animal.say();
        animal = new Cat();
        animal.say();
        Cat cat = new Cat();
        cat.eat();
    }
}

class Animal{
    public void say(){
        System.out.println("Animal say~"); 
    }
}

class Dog extends Animal{
    public void say(){
        System.out.println("Dog say~"); 
    }
}

class Cat extends Animal{
    public void say(){
        System.out.println("Cat say~"); 
    }

    public void eat(){
        System.out.println("Cat eat~");
    }
}

File: python3/oop_base/extends/duotai.py
class Animal(object):
    def run(self):
        print('Animal is runing...')

    def run_twice(self):
        self.run()

class Dog(Animal):
    def run(self):
        print('Dog is running...')

class Cat(Animal):
    def run(self):
        print('Cat is running...')

    def eat(self):
        print('Cat eat fish')

def run_twice(animal):
    animal.run()

dog = Dog()
cat = Cat()

dog.run_twice()
cat.run_twice()

run_twice(dog)
run_twice(cat)
